File: sandbox/algoritms/selectionSort.js
function findSmallest(arr) {
  let smallest = arr[0];
  let smallestIndex = 0;

  for (let i = 1; i < arr.length; i++) {
    if (arr[i] < smallest) {
      smallest = arr[i];
      smallestIndex = i;
    }
  }

  return smallestIndex;
}


function selectionSort(arr) {
    let newArr = [];
    
    // While there are elements in the array arr
    while (arr.length > 0) {
      // Find the index of the smallest element
      let smallest = findSmallest(arr);
      // Remove the smallest element and add it to newArr
      newArr.push(arr.splice(smallest, 1)[0]);
    }
  
    return newArr;
  }

console.log(selectionSort([5, 3, 6, 2, 10])); // [2, 3, 5, 6, 10]

File: Projects/Testtrack/algoritms/selection_sort.py
def _find_smallest(arr: list):
    if not arr:
        return None

    smallest = arr[0]
    smallest_index = 0

    for i, j in enumerate(arr):
        if (j < smallest):
            smallest = j
            smallest_index = i

    return smallest_index


def selection_sort(arr: list):
    new_arr = []
    arr_copy = arr.copy()

    while (len(arr_copy) > 0):
        smallest = _find_smallest(arr_copy)
        new_arr.append(arr_copy.pop(smallest))

    return new_arr


if __name__ == "__main__":
    print(selection_sort([5, 3, 6, 2, 10]))
